modules/services/colima.nix

```
@@ -0,0 +1,327 @@
+            description = "Enable nested virtualization.";
+          };
+          mountType = mkOption {
+            type = types.enum [
+              "virtiofs"
+              "9p"
+              "sshfs"
+            ];
+            description = "Volume mount driver for the virtual machine.";
+          };
+          mountInotify = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Propagate inotify file events to the VM.";
+          };
+          cpuType = mkOption {
+            type = types.str;
+            default = "";
+            description = "The CPU type for the virtual machine.";
+          };
+          provision = mkOption {
+            type = types.listOf (
+              types.submodule {
+                options = {
+                  mode = mkOption {
+                    type = types.enum [
+                      "system"
+                      "user"
+                    ];
+                    description = "Mode for the provision script.";
+                  };
+                  script = mkOption {
+                    type = types.str;
+                    description = "The provision script.";
+                  };
+                };
+              }
+            );
+            default = [ ];
+            description = "Custom provision scripts for the virtual machine.";
+          };
+          sshConfig = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Modify ~/.ssh/config automatically.";
+          };
+          sshPort = mkOption {
+            type = types.int;
+            default = 0;
+            description = "The port number for the SSH server.";
+          };
+          mounts = mkOption {
+            type = types.listOf (
+              types.submodule {
+                options = {
+                  location = mkOption {
+                    type = types.str;
+                    description = "Location to mount.";
+                  };
+                  writable = mkOption {
+                    type = types.bool;
+                    default = true;
+                    description = "Whether the mount is writable.";
+                  };
+                };
+              }
+            );
+            default = [ ];
+            description = "Configure volume mounts for the virtual machine.";
+          };
+          diskImage = mkOption {
+            type = types.str;
+            default = "";
+            description = "Specify a custom disk image.";
+          };
+          env = mkOption {
+            type = types.attrsOf types.str;
+            default = { };
+            description = "Environment variables for the virtual machine.";
+          };
+        };
+      };
+      default = { };
+      description = "Colima configuration settings.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    home.file.".colima/default/colima.yaml" = {
```

```suggestion
    home.file.".colima/default/colima.yaml" = lib.mkIf (cfg.settings != {}) {
```

If someone doesn't customize it, we shouldn't generate a config.